### ISSUE TITLE
compose control network domain name

### ISSUE THREAD

**Issue by dwaite** [docker/compose#3541]
When I create a "foo.com" network with docker-compose, the network appears to be changed and to be dynamic.

The containers themselves talk to each other with TLS, and do hostname validation. So, the DNS resolution for my network has to have the services be addressed in the form of "foo.bar.com".

Today I'm handling this by having docker-compose attach to an externally created network so that I can control the FQDN. Is there another way to accomplish this?

**Comment by shin-** [comment]
Are you looking for [`aliases`](https://docs.docker.com/compose/compose-file/#aliases)?

**Comment by dwaite** [comment]
Are aliases within a domain, or can they be FQDNs?

**Comment by shin-** [comment]
They're always fully qualified.

**Comment by dwaite** [comment]
I believe my issue is that for:
``` yml
networks:
  bar.com:
services:
  foo:
    networks:
      bar.com:
```
a FQDN is created based on the service label (not dynamic container name) and the dynamic network name (not network label). So my service isn't reachable as `foo.bar.com`, but something like `foo.folder_bar.com`.

Would it be worth opening an issue that there should be a default alias created of `foo.bar.com`?

**Comment by shin-** [comment]
This should work, unless I'm misunderstanding the use case.
```
version: "2"
networks:
  bar.com:
services:
  foo:
    networks:
      bar.com:
        aliases:
          - foo.bar.com
```

**Comment by dwaite** [comment]
Yes, it does. I'm wondering if I should open an issue that that particular alias should be automatic under compose - today without specifying an alias, your service will wind up with a DNS name in the form of foo.foldername_bar.com

**Comment by WolfgangFahl** [comment]
This report is more than a year old and for me falls in to the category of useability issues. The workaround above is non intuitive and most awkward. Setting the fqdn foo.bar.com directly would IMHO be most sensible.

**Comment by WolfgangFahl** [comment]
Again automatically closed with no fix. The need is still there - how do you no that it's not?

**Comment by glours** [comment]
We added the bot to do some automatic cleanup & get more visibility on what's really matter or not.
There's currently too much issues open, some for years, and it can be difficult to prioritize what to do.

When the bot mark an issue as stale, you only have to add a comment to remove this status.

I reopen the issue

**Comment by WolfgangFahl** [comment]
How about teaching the bot not to close issues that have a certain comment content /number of upvotes or the like?

**Comment by rbuckland** [comment]
Where is this ticket up to ? 
I would expect that 

```
  foobar:
    image: ${image_foobar}
    domainname: test-local
    dns_search:
            - test-local
```
would have expected
1. give the foobar image a DNS entry of `foobar` and `foobar.test-local` in the internal DNS server. It doesn't
2. to give all containers the default `service-name` and `service.${COMPOSE_PROJECT_NAME}`

the latter would mean simply 

```
# test-local/docker-compose.yml
...
  foobar:
    image: ${image_foobar}
```
automatically gives the entries `foobar` and `foobar.test-local`
This would make creating test harnesses really nice.

**Comment by dwaite** [comment]
Still a usability issue with a non-intuitive workaround. It's only stale in the sense that there's no proper triage for issues, so as such it may remain a usability issue forever.
